src/printer.js

```diff
@@ -133,31 +180,74 @@ function printMemberChain(path, options, print) {
   //    PropertyLookup d, Call]
   const printedNodes = [];
 
-  function traverse(path) {
+  // Here we try to retain one typed empty line after each call expression or
+  // the first group whether it is in parentheses or not
+  function shouldInsertEmptyLineAfter(node) {
+    const { originalText } = options;
+    const nextCharIndex = getNextNonSpaceNonCommentCharacterIndex(
+      originalText,
+      node,
+      options
+    );
+    const nextChar = originalText.charAt(nextCharIndex);
+
+    // if it is cut off by a parenthesis, we only account for one typed empty
+    // line after that parenthesis
+    if (nextChar === ")") {
+      return isNextLineEmptyAfterIndex(
+        originalText,
+        nextCharIndex + 1,
+        options
+      );
+    }
+
+    return isNextLineEmpty(originalText, node, options);
+  }
+
+  function rec(path) {
     const node = path.getValue();
-    if (node.kind === "call") {
+    if (
+      node.kind === "call" &&
+      (isMemberish(node.what) || node.what.type === "call")
+    ) {
       printedNodes.unshift({
         node,
-        printed: concat([printArgumentsList(path, options, print)])
+        printed: concat([
+          printArgumentsList(path, options, print),
+          shouldInsertEmptyLineAfter(node) ? hardline : ""
+        ])
       });
-      path.call(what => traverse(what), "what");
-    } else if (node.kind === "propertylookup") {
+      path.call(what => rec(what), "what");
+    } else if (isMemberish(node)) {
+      let printedMemberish = null;
+      if (node.kind === "propertylookup") {
+        printedMemberish = printPropertyLookup(path, options, print);
```

Comment:
maybe add a comment explaining why we do this instead of just calling `print` - im assuming its to avoid grouping?

Reply:
@mgrip hm, for me everything is clear, whatever you comment added?

Reply:
`"propertylookup", "staticlookup"`, etc are handled by `print` already - so whats the benefit to calling `printPropertyLookup` instead of just `path.call(print, "offset")`?

Reply:
@mgrip better output method/property chains :smile:

Reply:
ya so maybe a comment saying how its different and why its better 🤷‍♂️